API-1372: Doc Events API Format (#439)
akeneo/pim-api-docs: content/events-api/overview.md

## Scope of the Events API

### Events dispatched as JSON

When an event in your subscription occurs, we'll send an HTTP POST request to your Request URL. 

The event will be in the `Content-Type: application/json` format:

```json
{
    "events": [
        {
            "action": "product.created",
            "author": "julia",
            "author_type": "ui",
            "event_id": "6ad821d8-2468-4632-89d2-92e442c72313",
            "event_date": "2020-01-01T00:00:00+00:00",
            "pim_source": "staging.akeneo.com",
            "data": {
                "resource": {...}
            }
        }
    ]
}
```

### Request header

| Name | Example | Format |
| ---- | ------- | ------ |
| x-akeneo-request-signature | `d4582dd13a7b654c212d70b4c705e0ab0c0e6eb7f6524080f4d39121f88e6061` | |
| x-akeneo-request-timestamp | `1602520454` | Unix timestamp in seconds |

### Request body

| Name | Description | Example | Format |
| ---- | ----------- | ------- | ------ |

> | action | The action attribute helps you identify the event type you received, which action has been done in the PIM. | `product.created` | `[resource_name].[event_name]` |

Reviewer: ```suggestion
| action | The action property helps you identify the API event type you received, which action has been done in the PIM. | `product.created` | `[resource_name].[event_name]` |
```

> | event_id | The event_id is the unique identifier of the event. It allows you to be sure that you received the same event only one time. | `6ad821d8-2468-4632-89d2-92e442c72313` | UUID V4|

Reviewer: ```suggestion
| event_id | The event_id is the unique identifier of the event. It allows you to ensure you received the same API event only one time. | `6ad821d8-2468-4632-89d2-92e442c72313` | UUID V4|
```

> | event_date | The event_date contains the date and time of the event. | 2020-01-01T00:00:00+00:00 | Format ISO 8601 |

Reviewer: So that may be better to rename it event_datetime?

Author: Yes, done in https://github.com/akeneo/pim-api-docs/pull/439/commits/dddc3eee107bbe8d7585fbf88ae094b90565c6b1

> | author | The author attribute tells you who performed the action. | `julia` or `magento_0000` | |

Reviewer: ```suggestion
| author | The author property tells you who performed the action. | `julia` or `magento_0000` | |
```

> | author_type | The author_type is the type of user who performed the action. It can be a UI user, like Julia, or a REST API call coming from another application. | `ui` or `api` | |

Reviewer: What about backend tasks like rule engine?

Author: All non-API users will have author_type = 'UI' so, backend tasks will have 'UI' as author_type.

> | pim_source | The pim_source attribute helps you to identify which Akeneo PIM sent the event. | `staging.akeneo.com` | From `AKENEO_PIM_URL` environment variable |

Reviewer: ```suggestion
| pim_source | The pim_source property helps you to identify which Akeneo PIM sent the event. | `staging.akeneo.com` | From `AKENEO_PIM_URL` environment variable |
```
